pkg/glusterfs/controllerserver.go

```diff
@@ -63,44 +66,54 @@ type ProvisionerConfig struct {
 	// csiConf  *CsiDrvParam
 }
 
+func setProvisionerType(reqConf *ProvisionerConfig, value string) error {
+	if value != brickProvisionerLoop && value != brickProvisionerLvm {
+		return errors.New("invalid brick provisioner type")
+	}
+	reqConf.gdVolReq.ProvisionerType = value
+	return nil
+}
+
 // ParseCreateVolRequest parse incoming volume create request and fill
 // ProvisionerConfig.
 func (cs *ControllerServer) ParseCreateVolRequest(req *csi.CreateVolumeRequest) (*ProvisionerConfig, error) {
 
 	var (
 		reqConf ProvisionerConfig
 		gdReq   api.VolCreateReq
+		err     error
 	)
 
 	reqConf.gdVolReq = &gdReq
 
 	reqConf.gdVolReq.ReplicaCount = defaultReplicaCount
 	// Get Volume name
 	reqConf.gdVolReq.Name = req.Name
+	// Brick Provisioner Type
+	reqConf.gdVolReq.ProvisionerType = defaultBrickProvisionerType
 
 	for k, v := range req.GetParameters() {
 		switch k {
 		case "replicas":
 			replicas := v
-			replicaCount, err := parseVolumeParamInt(replicas)
-			if err != nil {
-				return nil, fmt.Errorf("invalid value for parameter '%s', %v", k, err)
-			}
-			reqConf.gdVolReq.ReplicaCount = replicaCount
+			reqConf.gdVolReq.ReplicaCount, err = parseVolumeParamInt(k, replicas)
 		case "arbiterType":
 			if v == "thin" {
-				if err := addThinArbiter(reqConf.gdVolReq, req); err != nil {
-					return nil, err
-				}
+				err = addThinArbiter(reqConf.gdVolReq, req)
 			}
 		case "arbiterPath":
 			if _, ok := req.Parameters["arbiterType"]; !ok {
 				glog.Error("only arbiterPath provided, missing arbiterType")
 			}
 			//skip incase of arbiterPath and arbiterType are provided
+		case "provisionerType":
+			err = setProvisionerType(&reqConf, v)
 		default:
 			glog.Errorf("invalid option specified: %s:%s", k, v)
 		}
+		if err != nil {
```

Comment:
as above one is a switch case, the error value may get overridden in different cases.